simulator/func_sim/alu.h

```diff
@@ -267,8 +267,8 @@ struct ALU
 
     template<typename T> static void add_uw( Instr* instr) { instr->v_dst[0] = instr->v_src[1] + ( bitmask<T>(32) & instr->v_src[0]); }
     template<typename T> static void bclr  ( Instr* instr) { instr->v_dst[0] = instr->v_src[0] & ~( lsb_set<T>() << shamt_v_src2<T>( instr)); }
-
     template<typename T> static void bseti( Instr* instr)  { instr->v_dst[0] = instr->v_src[0] | (lsb_set<T>() << (shamt_imm (instr) & (XLEN - 1))); }
+    template<typename T> static void sext_b( Instr* instr) { instr->v_dst[0] = instr->v_src[0] & 0x80 ? ~(msb_set<T>() | bitmask<T>(8)) | instr->v_src[0] : (msb_set<T>() | bitmask<T>(8)) & instr->v_src[0]; }
```

Comment:
Should we use `sign_extension( T value, size_t bits)` from macro.h?
https://github.com/MIPT-ILab/mipt-mips/blob/c2796c3c28bbd07438a019974efc765f6d589ce3/simulator/infra/macro.h#L57-L65